src/Planner/Planner.cpp

```diff
@@ -216,12 +216,17 @@ class QueryAnalysisResult
             limit_length = query_node.getLimit()->as<ConstantNode &>().getValue().safeGet<UInt64>();
         }
 
+        if (settings.limit)
```

Comment:
This will not work properly in case of distributed queries. Because it relies that we will send context with specific settings, and it does not look reliable.